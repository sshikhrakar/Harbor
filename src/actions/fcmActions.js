import {
  REGISTER_FCM_TOKEN,
  REGISTER_FCM_TOKEN_FULFILLED,
  REGISTER_FCM_TOKEN_ERRORED,
  REFRESH_FCM_TOKEN,
  REFRESH_FCM_TOKEN_FULFILLED,
  REFRESH_FCM_TOKEN_ERRORED,
} from './actionTypes';

/**
 * Dispatch FCM token registration action.
 *
 * @returns {Object}
 */
export function registerFcmToken() {
  return {
    type: REGISTER_FCM_TOKEN,
  };
}


/**
 * Dispatch when FCM token is successfully registered.
 *
 * @returns {Object}
 */
export function registerFcmTokenFulfilled() {
  return {
    type: REGISTER_FCM_TOKEN_FULFILLED,
  };
}

/**
 * Dispatch when FCM token registration fails.
 *
 * @param {Object} err
 * @returns {Object}
 */
export function registerFcmTokenErrored(err) {
  return {
    type: REGISTER_FCM_TOKEN_ERRORED,
    payload: {
      err,
    },
  };
}


/**
 * Action creator for failed refresh of FCM token.
 *
 * @param {String} token
 * @returns {Object}
 */
export function refreshFcmToken(token) {
  return {
    type: REFRESH_FCM_TOKEN,
    payload: {
      token,
    },
  };
}

/**
 * Action creator for failed refresh of FCM token.
 *
 * @returns {Object}
 */
export function refreshFcmTokenErrored() {
  return {
    type: REFRESH_FCM_TOKEN_ERRORED,
  };
}

/**
 * Action creator for successful refresh of FCM token.
 *
 * @returns {Object}
 */
export function refreshFcmTokenFulfilled() {
  return {
    type: REFRESH_FCM_TOKEN_FULFILLED,
  };
}
