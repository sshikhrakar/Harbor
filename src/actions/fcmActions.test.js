import * as fcmActions from './fcmActions';

describe('ACTION CREATORS: FCM', () => {
  it('should create registration action', () => {
    expect(fcmActions.registerFcmToken('foobarbaz1234foobarfaz1234')).toMatchSnapshot();
  });

  it('should create registration success action', () => {
    expect(fcmActions.registerFcmTokenFulfilled()).toMatchSnapshot();
  });

  it('should create registration failure action', () => {
    expect(fcmActions.registerFcmTokenErrored()).toMatchSnapshot();
  });

});
