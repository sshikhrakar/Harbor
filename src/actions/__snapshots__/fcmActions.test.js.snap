// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACTION CREATORS: FCM should create registration action 1`] = `
Object {
  "payload": Object {
    "token": "foobarbaz1234foobarfaz1234",
  },
  "type": "FCM/REGISTER_FCM_TOKEN",
}
`;

exports[`ACTION CREATORS: FCM should create registration failure action 1`] = `
Object {
  "payload": Object {
    "err": undefined,
  },
  "type": "FCM/REGISTER_FCM_TOKEN_ERRORED",
}
`;

exports[`ACTION CREATORS: FCM should create registration success action 1`] = `
Object {
  "type": "FCM/REGISTER_FCM_TOKEN_FULFILLED",
}
`;
